shared/js/language_list.js

```diff
@@ -111,13 +121,18 @@ exports.LanguageList = {
       this._readSetting('deviceinfo.os'),
       this._readSetting('language.current'),
       this._readSetting('devtools.qps.enabled'),
+      this._readSetting('accessibility.screenreader'),
       navigator.mozApps.getAdditionalLanguages()
-    ]).then(function([langsFromFile, ver, current, qpsEnabled, addl]) {
-      var langs = this._copyObj(langsFromFile);
-      this._extendPseudo(langs, current, qpsEnabled);
-      this._extendAdditional(langs, this._parseVersion(ver), addl);
-      return [langs, current];
-    }.bind(this));
+    ]).then(
+      function([langsFromFile, ver, current, qpsEnabled, srEnabled, addl]) {
+        var langs = this._copyObj(langsFromFile);
+        this._extendPseudo(langs, current, qpsEnabled);
+        this._extendAdditional(langs, this._parseVersion(ver), addl);
+        if (srEnabled) {
+          this._removeWithNoSpeech(langs);
```

Comment:
Just looking at the other functions here, could you pass srEnabled as an arg and check if it's enabled internally?